plugins/woocommerce-blocks/assets/js/blocks/product-gallery/frontend.tsx

```diff
@@ -30,92 +28,41 @@ const getContext = ( ns?: string ) =>
 type Store = typeof productGallery & StorePart< ProductGallery >;
 const { state, actions } = store< Store >( 'woocommerce/product-gallery' );
 
-/**
- * Product Gallery supports two contexts:
- * - on-page gallery - may display subset of images.
- * - dialog gallery - displays all of the images.
- * Function returns images per current context.
- */
-const getCurrentImages = ( context: ProductGalleryContext ) => {
-	const { isDialogOpen } = context;
-	return context[
-		isDialogOpen ? 'dialogVisibleImagesIds' : 'visibleImagesIds'
-	];
+const getImageIndex = (): number => {
```

Comment:
Feels like this should belong on the state object with the other getters? What do you think?

Reply:
I did that but then, I couldn't use it in another getter as it's been telling me there's circular dependency... Which I don't get as we derive state from other state properties in some other getters. I left it in such a state: https://github.com/woocommerce/woocommerce/pull/55042/commits/290fa5f02852d76335fee9e2535acf255f36bb9b